docs/specifications/tests/Nameserver-TP/nameserver11.md

```diff
@@ -3,91 +3,142 @@
 
-## Ordered description of steps to be taken to execute the test case
+## Summary
+
+Message                                | Level     | Arguments           | Description of when message tag is outputted
+:--------------------------------------|:----------|---------------------|---------------------------------------------
+NS11_UNKNOWN_OPTION_CODE               | WARNING   | ns_ip_list          | The DNS response contains an unknown EDNS option-code
+
+The value in the Level column is the default severity level of the message. The
+severity level can be changed in the [Zonemaster-Engine profile]. Also see the
+[Severity Level Definitions] document.
+
+The argument names in the Arguments column lists the arguments used in the
+message. The argument names are defined in the [argument list].
+
+
+## Test procedure
+
+In this section and unless otherwise specified below, the term "[DNS Query]"
+follows the specification for DNS queries as specified in
+[DNS Query and Response Defaults]. The handling of the DNS responses on the DNS
+queries follow, unless otherwise specified below, what is specified for
+[DNS Response] in the same specification.
 
-1. Create a SOA query for the *Child Zone* with an OPT record with 
-   EDNS OPTION-CODE set to anything other than it is already assigned as in the
-[IANA-DNSSYSTEM-PARAMETERS] and no other EDNS options or flags.
+1. Create the following empty sets:
+   1. Name server IP address ("Unknown Option Code")
 
-2. Obtain the set of name server IP addresses using [Method4] and [Method5]
+2. Create an SOA query for the *Child Zone* with an OPT record with 
+   EDNS OPTION-CODE set to anything other than it is already assigned as in
+   the [IANA-DNSSYSTEM-PARAMETERS] and no other EDNS options or flags.
+
+3. Obtain the set of name server IP addresses using [Method4] and [Method5] 
    ("Name Server IP").
 
-3. For each name server in *Name Server IP* do:
-
-	1. Send the SOA query to the name server and collect the response.
-	2. If there is no DNS response, output *[NO_RESPONSE]* and go to
-	next server.
-	3. Else, if the DNS response has the RCODE "FORMERR" then output
-      	*[NO_EDNS_SUPPORT]*.
-	4. Else, if there is an "OPTION-CODE" present in the response, then
-	output *[UNKNOWN_OPTION_CODE]*. 
-	5. Else, if the DNS response meet the following four criteria,
-      	then just go to the next name server (no error):
-		1. The SOA is obtained as response in the ANSWER section.
-		2. If the DNS response has the RCODE "NOERROR".
-		3. The pseudo-section response has an OPT record with version set to 0.
-		4. There is no "OPTION-CODE" present in the response.
-	6. Else output *[NS_ERROR]*.
- 
+4. For each name server in *Name Server IP* do:
+
+   1. Send the SOA query to the name server and collect the response.
+   2. If there is no DNS response, then go to next name server.
+   3. Else, if the DNS response has the RCODE "FORMERR" ([IANA RCODE List]), then go to next name server.
```

Comment:
I suggest that we write "has the [RCODE Name] FormErr" with [RCODE Name] being a link to IANA and use the same form of RCODE name as in the list.

Reply:
Fixed